src/commands/nsec3hash.rs

```diff
@@ -111,71 +117,274 @@ impl Nsec3Hash {
+
+        #[test]
+        fn accept_good_cli_args() {
+            assert_cmd_eq(&["nlnetlabs.nl"], "e3dbcbo05tvq0u7po4emvbu79c8vpcgk.\n");
+            assert_cmd_eq(
+                &["-a", "1", "nlnetlabs.nl"],
+                "e3dbcbo05tvq0u7po4emvbu79c8vpcgk.\n",
+            );
+            assert_cmd_eq(
+                &["-t", "0", "nlnetlabs.nl"],
+                "asqe4ap6479d7085ljcs10a2fpb2do94.\n",
+            );
+            assert_cmd_eq(
+                &["-t", "1", "nlnetlabs.nl"],
+                "e3dbcbo05tvq0u7po4emvbu79c8vpcgk.\n",
+            );
+            assert_cmd_eq(
+                &["-s", "", "nlnetlabs.nl"],
+                "e3dbcbo05tvq0u7po4emvbu79c8vpcgk.\n",
+            );
+            assert_cmd_eq(
+                &["-s", "DEADBEEF", "nlnetlabs.nl"],
+                "2h8rboqdrq0ard25vrmc4hjg7m56hnhd.\n",
+            );
+        }
+
+        #[test]
+        fn reject_bad_cli_args() {
+            assert!(parse_cmd_line(&[]).is_err());
+            assert!(parse_cmd_line(&[""]).is_err());
+
+            assert!(parse_cmd_line(&["-a"]).is_err());
+            assert!(parse_cmd_line(&["-a", "nlnetlabs.nl"]).is_err());
+            assert!(parse_cmd_line(&["-a", "", "nlnetlabs.nl"]).is_err());
+            assert!(parse_cmd_line(&["-a", "2", "nlnetlabs.nl"]).is_err());
+            assert!(parse_cmd_line(&["-a", "SHA1", "nlnetlabs.nl"]).is_err());
+            assert!(parse_cmd_line(&["-a", "SHA-1", "nlnetlabs.nl"]).is_err());
+            assert!(parse_cmd_line(&["-a", "SHA-256", "nlnetlabs.nl"]).is_err());
+
+            assert!(parse_cmd_line(&["-t"]).is_err());
+            assert!(parse_cmd_line(&["-t", "nlnetlabs.nl"]).is_err());
+            assert!(parse_cmd_line(&["-t", "", "nlnetlabs.nl"]).is_err());
+            assert!(parse_cmd_line(&["-t", "-1", "nlnetlabs.nl"]).is_err());
+            assert!(parse_cmd_line(&["-t", "abc", "nlnetlabs.nl"]).is_err());
+            assert!(
+                parse_cmd_line(&["-t", &((u16::MAX as u32) + 1).to_string(), "nlnetlabs.nl"])
+                    .is_err()
+            );
+
+            assert!(parse_cmd_line(&["-s"]).is_err());
+            assert!(parse_cmd_line(&["-s", "nlnetlabs.nl"]).is_err());
+            assert!(parse_cmd_line(&["-s", "NOTHEX", "nlnetlabs.nl"]).is_err());
+            assert!(parse_cmd_line(&["-s", &"aa".repeat(256), "nlnetlabs.nl"]).is_err());
+        }
+
+        //------------ Helper functions ------------------------------------------
+
+        fn parse_cmd_line(args: &[&str]) -> Result<Args, Error> {
+            parse_args(|| ["ldns-nsec3-hash"].iter().chain(args).map(Into::into))
+        }
+
+        fn assert_cmd_eq(args: &[&str], expected_output: &str) {
+            let parsed_args = parse_cmd_line(args).unwrap();
+            let mut captured_stdout = vec![];
+            parsed_args.execute(&mut captured_stdout).unwrap();
+            assert_eq!(str::from_utf8(&captured_stdout), Ok(expected_output));
```

Comment:
This is what this would look like after #23:
```rust
let env = FakeEnv {
    args: ["ldns-nsec3-hash"].iter().chain(args).map(Into::into)).collect(),
    ..Default::default(),
}

let res = run(&env);
assert_eq!(res, 0);
assert_eq!(env.get_stdout(), expected_output);
assert_eq!(env.get_stderr(), expected_output);
```